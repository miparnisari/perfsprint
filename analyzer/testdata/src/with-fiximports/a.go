package a

func loopconcat() { // want "fiximports: Fix imports"
	// no imports, adding strings
	s := ""
	for i := 0; i < 10; i++ {
		s += "toto" // want "concat-loop: string concatenation in a loop"
	}
	println("Hello, World!", s)
}
