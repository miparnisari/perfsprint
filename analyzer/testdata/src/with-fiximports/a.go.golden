package a

import (
	"strings"
)

func loopconcat() { // want "fiximports: Fix imports"
	// no imports, adding strings
	s := ""
	var sSb6 strings.Builder
	for i := 0; i < 10; i++ {
		sSb6.WriteString("toto") // want "concat-loop: string concatenation in a loop"
	}
	s += sSb6.String()
	println("Hello, World!", s)
}
